tfjs-backend-webgpu/src/kernels/gather_nd_webgpu.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {util} from '@tensorflow/tfjs-core';
+
+import {getCoordsDataType} from '../shader_preprocessor';
+import {computeDispatch, flatDispatchLayout} from '../webgpu_util';
+
+import {WebGPUProgram} from './webgpu_program';
+
+export class GatherNDProgram implements WebGPUProgram {
+  outputShape: number[];
+  shaderKey: string;
+  dispatchLayout: {x: number[]};
+  dispatch: [number, number, number];
+  variableNames: string[] = ['A', 'indices'];
+  uniforms: string;
+  workGroupSize: [number, number, number] = [64, 1, 1];
+  size: number;
+  sliceDim: number;
+  constructor(sliceDim: number, strides: number[], shape: number[]) {
+    this.outputShape = shape;
+    this.dispatchLayout = flatDispatchLayout(this.outputShape);
+    this.dispatch = computeDispatch(
+        this.dispatchLayout, this.outputShape, this.workGroupSize);
+    this.shaderKey = `gathernd_${sliceDim}`;
+    this.size = util.sizeFromShape(this.outputShape);
+    this.sliceDim = sliceDim;
+    this.uniforms =
+        `int sliceDim; ${getCoordsDataType(strides.length)} strides;`;
```

Comment:
Is `sliceDim` equal to `strides.length`? If yes, you can remove the parameter `strides: number[]` .

Reply:
It seems that it is. See https://github.com/tensorflow/tfjs/blob/master/tfjs-core/src/ops/gather_nd_util.ts#L82. Then you can write it like below:
```
constructor(sliceDim: number, shape: number[])
...
this.uniforms =
        `int sliceDim; ${getCoordsDataType(sliceDim)} strides;`;